service/history/hsm/tree.go

```diff
@@ -90,17 +96,90 @@ type cachedMachine struct {
 	deleted bool
 }
 
-type OperationLog struct {
-	// All transitions, including regular state changes and deletions
-	TransitionsByPath map[string][]TransitionOutputWithCount
-	// Track topmost deleted paths to filter outputs efficiently
-	DeletedPaths [][]Key
+// String returns a human-readable representation of a path
+func formatPath(path []Key) string {
+	if len(path) == 0 {
+		return "<root>"
+	}
+	var parts []string
+	for _, key := range path {
+		parts = append(parts, key.String())
+	}
+	return strings.Join(parts, "/")
```

Comment:
This would be ambiguous if the keys contained slashes but it's fine, it's just for debugging.